provisioning/image_cache.go

```diff
@@ -77,9 +76,32 @@ func imageCacheConfig(targetNamespace string, config metal3iov1alpha1.Provisioni
 		Scheme: "http",
 		Host: net.JoinHostPort(fmt.Sprintf("%s.%s.svc.cluster.local", stateService, targetNamespace),
 			baremetalHttpPort),
-		Path: fmt.Sprintf("/images/%s/%s", imageName, imageName),
+		Path: fmt.Sprintf("/images/%s/cached-%s", imageName, imageName),
```

Comment:
Is this correct for both qcow2 and live-iso? I would guess that it's not returning the correct thing when there is no -live image, because in that case the machine-os-downloader container is still running the old `get-resource.sh` and not the new `get_live_images.sh`.

Also, this doesn't match the comment on line 73, so please update it.

Reply:
The filename prefixed with `cached` is the original name. The one without is a symlink. So, this should be OK.

https://github.com/openshift/ironic-rhcos-downloader/blob/d4f21a6e47399ae796a7c0ea80c211b8405d0fb1/get-resource.sh#L104

Reply:
Ahh, I see the problem. Yes, this will fail spectacularly, if we always prefix with `cache-`

Reply:
🤦 :
```
sh-4.4# ls -l /shared/html/images/cached-rhcos-48.84.202105190318-0-live-initramfs.x86_64.img/
total 85944
-rw-r--r--. 1 root root 88003520 Jul 19 23:16 cached-cached-rhcos-48.84.202105190318-0-live-initramfs.x86_64.img
```

Reply:
I'll open a PR on ironic-rhcos-downloader and add the symlink for the live images as well. That should fix this problem. Thanks for catching this one and clarifying it for me.

Reply:
PR here: https://github.com/openshift/ironic-rhcos-downloader/pull/59